### PR TITLE
#MySQL_Tasks. Ilya Maylychko

### PR DESCRIPTION
**Pull request by juggler39**
https://travis-ci.org/github/juggler39/db-assignments/builds/711042331

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -106,7 +161,17 @@ async function task_1_6(db) {
...
  */
 async function task_1_7(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        first.EmployeeID AS EmployeeId,
+        CONCAT(first.FirstName, ' ', first.LastName) AS FullName,
+        IFNULL(CONCAT(second.FirstName, ' ', second.LastName), '-') AS ReportsTo
+    FROM
+        Employees first
```
Comment (AlexMokin) [inline task/sql-tasks.js]: figure out better aliases. 'first' and 'second' are bad aliases. If the query is hsort just use first letter or two from the table name

File: task/sql-tasks.js [hunk 2]
```diff
@@ -178,7 +291,14 @@ async function task_1_12(db) {
  *
  */
 async function task_1_13(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        COUNT(ProductID) AS TotalOfCurrentProducts,
+        SUM(Discontinued) AS TotalOfDiscontinuedProducts
```
Comment (AlexMokin) [inline task/sql-tasks.js]: clever. Although for readability it's better to write **discontinued = 1** expression explicitly. 

e.g.
```
select
         (select Count(*) from Products where Discontinued = 1) as `TotalOfDiscontinuedProducts`,
	 (select Count(*) from Products) as `TotalOfCurrentProducts`;
```

File: task/sql-tasks.js [hunk 3]
```diff
@@ -200,7 +328,26 @@ async function task_1_14(db) {
  *
  */
 async function task_1_15(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        SUM(MONTH(OrderDate) = 1) AS January,
```
Comment (AlexMokin) [inline task/sql-tasks.js]: try to use count

File: task/sql-tasks.js (hunk 1 above)
Comment (juggler39) [reply task/sql-tasks.js]: OK. I replaced them with e and e2 accordingly.

File: task/sql-tasks.js (hunk 2 above)
Comment (juggler39) [reply task/sql-tasks.js]: I always undergo the spirit of 'code golf' by doing such exercises. I changed the code to your suggestions.

**Comment by juggler39** [comment]
Thank you for the review. I corrected all the notes.

https://travis-ci.org/github/juggler39/db-assignments/builds/721254514

File: task/sql-tasks.js (hunk 3 above)
Comment (juggler39) [reply task/sql-tasks.js]: Done

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**